src/schedule/edf_schedule.c

```diff
@@ -275,6 +275,9 @@ int scheduler_init_edf(void)
 	if (edf_sch->irq < 0)
 		return edf_sch->irq;
 
+	interrupt_mask(edf_sch->irq, cpu_get_id());
+	interrupt_unmask(edf_sch->irq, cpu_get_id());
```

Comment:
@bkokoszx @mwasko fyi - as discussed on Tuesday.
@lyakh is masking and unmasking flow optimal, should we not be clearing the IRQ with ```interrupt_clear()``` ? We also need a comment here explaining why we are doing this.

Reply:
@lgirdwood of course it isn't optimal and yes, `interrupt_clear()` *would be* the right function to call, but currently it only clears DSP "native" interrupts, and this is a level-2 interrupt. Let's test if this helps, if it does, we can try to extend `interrupt_clear()` to also work on "level interrupts."

Reply:
@lyakh I did the stress test on this PR. The bug https://github.com/thesofproject/sof/issues/4538 still can be reproduced. On the meantime, I did the stress test on the PR https://github.com/thesofproject/sof/pull/4874. The bug #4538 can be fixed and no other regression is found. So I will ask Dell team to test PR4874.

Reply:
> 

Thanks for the test. @lyakh @lgirdwood I would suggest to make the #4874 as hotfix for 1.9 and keep main unchanged as root causing the #4538 could be very time consuming.